File: Week04/Day-02/garden/flower.ts
import { Plant } from "./plant";

'use strict';

export class Flower extends Plant {
  constructor(color: string, type: string = 'flower', waterLevel: number = 0, needWater: boolean = true, waterAbsorb: number = 0.75) {
    super(color, type, waterLevel, needWater, waterAbsorb);
    if (color === 'yellow' || color === 'blue') {
      this.color = color;
    } else {
      throw new Error ('You can only choose yellow or blue flowers buddy.');
    }
    if (this.waterLevel <= 5) {
      this.needWater = true;
    } else {
      this.needWater = false;
    }
  }

  getFlowerColor() {
    if (this.color = 'yellow') {
      return 'yellow';
    } else {
      return 'blue'
    }
  }
}

File: Week10/garden-app/flower.py
from plant import Plant

class Flower(Plant):

    def __init__(self, color):
        super().__init__(color)
        self.water_absorb = 0.75

        if self.water_level < 5:
            self.need_water = True
        else:
            self.need_water = False

flower1 = Flower('yellow')
print(flower1.need_water)